custom-words.txt

```diff
@@ -1,31 +1,16 @@
-A128CBCPAD
-A192CBCPAD
-A256CBCPAD
-AADDS
```

Comment:
Why are you deleting anything from this file? I would expect your changes to only be additive, if necessary.

Reply:
Shouldn't be.  Weird.  I'll look into it.

Reply:
Looks to me that this is a diff "issue" with that branch we started from and what we are targeting.  The changes should only include our updates which were only additions.  Can I just go ahead and fix it up on this branch?

Reply:
It's best you rebase on main and force push. Simple merges get very messy:

```bash
# assuming upstream is Azure/azure-rest-api-specs and origin is a eqtip/azure-rest-api-specs fork (don't generally push to upstream and work in a fork instead)
git fetch upstream main
# on your published/Azure/azure-rest-api-specs-pr/RPSaaSMaster-easm-data-plane branch:
git rebase upstream/main
# fix any conflicts and follow on-screen instructions
git push --force
```

Reply:
@heaths So the problem here is that there are a number of changes coming in from our branch that are from its source branch that I believe we don't want applied to the public repo.  Specifically there are a TON of adds and removes we didn't make.  

By rebasing I could fix the conflicts but still ended up with said changes.  I instead created a branch that manually applied our changes and it's way more sensible:

https://github.com/Azure/azure-rest-api-specs/compare/main...eqtip:azure-rest-api-specs:published/Azure/azure-rest-api-specs-pr/RPSaaSMaster-easm-data-plane

Thoughts?

Reply:
Anything in main is published and you cannot make breaking changes. Whether you merge (messy) or rebase (clean), you still end up with the same result (rebasing is just easier to review). There should be nothing in main you don't want.

All rebasing does is like taking all your commits from a common ancestor and effectively "replays" them (not quite, but a close analogy) on main after updating your branch. See https://git-scm.com/book/en/v2/Git-Branching-Rebasing

Reply:
@heaths Thanks ... I'm familiar with rebasing just have dealt with conflicts in different ways for different projects in the past and wanted to get the lay of the land.  With that said, I've push my changes to my fork:

https://github.com/Azure/azure-rest-api-specs/compare/main...eqtip:azure-rest-api-specs:published/Azure/azure-rest-api-specs-pr/RPSaaSMaster-easm-data-plane

Thanks look good but where to from here?  Do I need to open up a separate PR?  As far as I can tell I'm not allowed to push changes to the branch in the Azure repo.

Reply:
You could force-push to this PR. But is this swagger considered _done_? The `main` branch is for publishing specs that, mostly, should be immutable after merging to `main`. See https://aka.ms/AzBreakingChangesPolicy for more information.

If not, others should make topic branches from this branch and submit PRs to it until your done, or make a dev- branch as described here: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/630/Walkthrough-Creating-a-new-service-specification-in-a-team

Reply:
Yes.  It is for this particular version.  We have another follow up version that went through the review process and will subsequently go through this same merge process.

Reply:
@heaths It doesn't look like I have permission to push to the PR branch.

Reply:
@heaths Any idea why I wouldn't have access to be able to push?  I've tried just about everything but no luck.  It should be noted I am not the original branch author.  Seems crazy that would be a limitation but maybe so?

Reply:
What error did you get? It's not a protected branch.

Reply:
@heaths Figured it out.  Was missing membership in a particular team.  Updates have been pushed so conflicts are resolved.